cloudquery/sdk/scalar/vector.py

```diff
@@ -0,0 +1,41 @@
+from .scalar import Scalar
+from typing import Type
+
+
+class Vector:
+    def __init__(self, type: Type[Scalar] = None, *args):
+        self.data = []
+        self.type = type
+        for arg in args:
+            self.append(arg)  # Use the append method for type checking and appending
+
+    def append(self, item):
+        if not isinstance(item, Scalar):
+            raise TypeError("Item is not of type Scalar or its subclass")
+
+        if self.type is None:
+            self.type = type(item)
+            self.data.append(item)
+        elif isinstance(item, self.type):
+            self.data.append(item)
+        else:
+            raise TypeError(f"Item is not of type {self.type.__name__}")
+
+    def __eq__(self, other):
+        if not isinstance(other, Vector):
+            return False
+        if len(self) != len(other):
+            return False
+        for self_item, other_item in zip(self.data, other.data):
+            if self_item != other_item:
+                return False
```

Comment:
This will be pretty slow and assumes equal order of items.
Is there a reason you're not using `collections.Counter`?
```
if Counter(self.data) != Counter(other.data):
    ...
```

Reply:
equal order of items is necessary, it's not a set type. Removed `zip`